### ISSUE TITLE
ClassNotFoundException: org.mockito.exceptions.Reporter

### ISSUE THREAD

**Issue by SebastianDietrich** [powermock/powermock#734]
powermock-api-mockito2 latest (1.6.6) refers to mockito-core 2.0.42-beta
Currently there is a mockito-core 2.3.7

Most people using mockito2 with powermock will _not_ use an outdated beta version of mockito2. Using maven they will have a different  version of mockito-core in their classpath. The API of mockito2 has changed since 2.0.42-beta - the class Reporter is no longer found in org.mockito.exceptions but in org.mockito.internal.exceptions.

--> please upgrade the dependency to the latest mockito-core 2 version

**Comment by thekingnothing** [comment]
I understand that most people want to use PowerMock with latest version of Mockito. But moving to Mockito 2 it's not just a change in dependency version and small changes in code. But it requires rewriting integration with Mockito, because PowerMock is heavily relying on the internal workings of Mockito (#726). 

You are welcome to help us with identify all places where internal Mockito API should be replaced with public API.

**Comment by thekingnothing** [comment]
Duplicates #684

**Comment by SebastianDietrich** [comment]
Sorry - I don't understand your answer - I am not talking abou tpowermock, but of powermock-api-mockito2 which already works with mockito 2

**Comment by thekingnothing** [comment]
powermock-api-mockito2 was attempt to follow mockito2 changes during development. It's a copy from powermock-api-mockito with some changes to fit mockito 2. What's why it depends on beta version.

I was trying to follow, but I failed. Later, I had discussion with Mockito team. We came to conclusion that the best way to provide support of Mockito 2 is use only public Mockito SPI and they ready to help PowerMock team and make changes in SPI.

Best regrads,
Arthur Zagretdinov


On 27 Dec 2016, at 14:25, SebastianDietrich <notifications@github.com<mailto:notifications@github.com>> wrote:


Sorry - I don't understand your answer - I am not talking abou tpowermock, but of powermock-api-mockito2 which already works with mockito 2

-
